pkg/controller/stack/resolver.go

```diff
@@ -108,3 +117,109 @@ func SearchStack(stackName string, index *Index) ([]Stack, error) {
 
 	return stackRefs, nil
 }
+
+// Returns true if the user specified values Kabanero.Spec.Stacks.Repositories.GitRelease.
+// Note that Kabanero.Spec.Stacks.Repositories.GitRelease.Hostname is excluded because
+// users may or may not specify it when connecting to a public Git repository.
+func isGitReleaseUsable(gitRelease kabanerov1alpha2.GitReleaseSpec) bool {
+	return len(gitRelease.Organization) != 0 && len(gitRelease.Project) != 0 &&
+		len(gitRelease.Release) != 0 && len(gitRelease.AssetName) != 0
+
+}
+
+// Retrieves a stack index file content using HTTP.
+func getStackIndexUsingHttp(repoConf kabanerov1alpha2.RepositoryConfig) ([]byte, error) {
+	url := repoConf.Https.Url
+	// user may specify url to yaml file or directory
+	matched, err := regexp.MatchString(`/([^/]+)[.]yaml$`, url)
+	if err != nil {
+		return nil, err
+	}
+	if !matched {
+		url = url + "/index.yaml"
+	}
+
+	return getFromCache(url, repoConf.Https.SkipCertVerification)
+}
+
+// Retrieves a stack index file content using Git APIs
+func getStackIndexUsingGit(repoConf kabanerov1alpha2.RepositoryConfig) ([]byte, error) {
+	var indexBytes []byte
+
+	// Get a Github client.
+	gclient, err := getGitClient(repoConf.GitRelease)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the release tagged in Github as repoConf.GitRelease.Release.
+	release, response, err := gclient.Repositories.GetReleaseByTag(context.Background(), repoConf.GitRelease.Organization, repoConf.GitRelease.Project, repoConf.GitRelease.Release)
+	if err != nil || response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Unable to retrieve object representing Github repository release %v. Configured GitRelease data: %v. Error: %v", repoConf.GitRelease.Release, repoConf.GitRelease, err)
```

Comment:
Should we report the HTTP status code here too?

Reply:
Yeah it would be helpful if it had that in the message.